File: worksheets/dp-four/scala/src/memento/CareTaker.scala
package memento

import java.util.HashMap
import java.util.Map

case class CareTaker() {

  private val savepointStorage: Map[String, Memento] =
    new HashMap[String, Memento]()

  def saveMemento(memento: Memento, savepointName: String): Unit = ???

  def getMemento(savepointName: String): Memento = ???

  def clearSavepoints(): Unit = ???
}

File: worksheets/dp-four/kotlin/src/memento/CareTaker.kt
package memento

class CareTaker() {

    private val savepointStorage = mutableMapOf<String, Memento>()

    fun saveMemento(memento: Memento, savepointName: String) {
        TODO()
    }

    fun getMemento(savepointName: String): Memento {
        TODO()
    }

    fun clearSavepoints() {
        TODO()
    }
}
